lib/purchases_flutter.dart

```diff
@@ -562,6 +562,15 @@ class Purchases {
   static Future<String> get appUserID async =>
       await _channel.invokeMethod('getAppUserID') as String;
 
+  /// Gets the current storefront for the store account.
+  static Future<Storefront?> get storefront async {
+    final storefrontJson = await _channel.invokeMethod('getStorefront');
```

Comment:
Question (since I'm not experienced on this): are all map/dictionary objects in the native transformed into JSON by Flutter?

Reply:
To be fair, it's not actually json (as in not a json string or an object representing a Json), but just a simple Dart Map yeah. The MethodChannel performs the transformation for those basic types. You can see those here: https://docs.flutter.dev/platform-integration/platform-channels#codec

Reply:
Oh, I see. Then, in the line below
```
return Storefront.fromJson(Map<String, dynamic>.from(storefrontJson));
```
if `storefrontJson` is already a Dart Map, why is
```
Map<String, dynamic>.from(storefrontJson)
```
needed?

(Sorry for the newbie questions 🙏)

Reply:
Not a newbie question at all!! So what we get from the channel is actually a “Map<Object?, Object?>”. We could try to parse this directly but since we know the type of this map, it’s easier to convert it to the expected types like this before actually parsing the map I think. There might be better ways to do this but this was the easiest IMO

Reply:
Ohh this makes sense. Thanks so much for explaining!